alacritty_terminal/src/term/mod.rs

```diff
@@ -355,16 +355,22 @@ impl<T> Term<T> {
         match self.selection.as_ref() {
             Some(Selection { ty: SelectionType::Block, .. }) => {
                 for line in (start.line.0..end.line.0).map(Line::from) {
-                    res += self
-                        .line_to_string(line, start.column..end.column, start.column.0 != 0)
-                        .trim_end();
+                    let s =
+                        &self.line_to_string(line, start.column..end.column, start.column.0 != 0);
 
-                    // If the last column is included, newline is appended automatically.
-                    if end.column != self.columns() - 1 {
+                    res += s;
+
+                    // Every line, except the last line, of block selection, must include a
+                    // trailing newline, but `line_to_string` may return a non-trailing newline
+                    // string when the selection doesn't contain the last column or it is wrapped.
+                    // So we need to append a newline if such a situation happens.
+                    if !s.ends_with('\n') {
```

Comment:
What's the point of the separate variable `s`?

Reply:
The reason is for correct string joining even if `s` is an empty string. Empty lines are discarded if not doing so.